src/test/java/org/cactoos/iterator/IteratorOfTest.java

```diff
@@ -39,45 +42,54 @@
 
     @Test
     public void emptyIteratorDoesNotHaveNext() {
-        MatcherAssert.assertThat(
-            "Can't create empty iterator",
+        new Assertion<>(
+            "must create empty iterator",
             new IteratorOf<>().hasNext(),
-            CoreMatchers.equalTo(false)
-        );
+            new IsNot<>(new IsTrue())
+        ).affirm();
     }
 
-    @Test(expected = NoSuchElementException.class)
+    @Test
     public void emptyIteratorThrowsException() {
-        new IteratorOf<>().next();
+        new Assertion<>(
+            "must throw an exception if empty",
+            () -> new IteratorOf<>().next(),
+            new Throws<>(NoSuchElementException.class)
+        ).affirm();
     }
 
     @Test
     public void nonEmptyIteratorDoesNotHaveNext() {
         final IteratorOf<Integer> iterator = this.iteratorWithFetchedElements();
```

Comment:
@victornoel According with [this Yegor's post](https://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html) tests must share nothing. So, can you leave a puzzle to refactor this? (it's interesting leave the link of the post too).